app/models/anonymous_user.rb

```diff
@@ -80,4 +80,20 @@ def casual_name
     full_name
   end
 
+  User.faculty_statuses.each do |status, value|
+    define_method "#{status}?" do
+      User::DEFAULT_FACULTY_STATUS.to_s == status
+    end
+
+    define_method "#{status}!" do
```

Comment:
You might want to raise an error here to make sure we don't attempt making an AnonymousUser faculty.